### ISSUE TITLE
Turkish small caps return wrong glyph

### ISSUE THREAD

**Issue by alerque** [psb1558/Elstob-font#38]
When the language is set to Turkish, the small caps feature should return a dotted-i glyph for i inputs. See this example were ElstobD (v2.201) fails to do so and returns an undotted-i shape instead as in English. The second example line here is Brill, which is correct:

```sile
\begin[papersize=a7]{document}
\nofolios
\font[family=ElstobD,language=en,features=+smcp]{First Input: }
\font[family=ElstobD,language=tr,features=+smcp]{Bir İlk Ilık}

\font[family=Brill,language=en,features=+smcp]{First Input: }
\font[family=Brill,language=tr,features=+smcp]{Bir İlk Ilık}
\end{document}
```

![20231104_16h02m14s_grim](https://github.com/psb1558/Elstob-font/assets/173595/c831c5b1-3505-45ba-b77d-c1d7b6258c78)

**Comment by alerque** [comment]
Come to think of it, the lowercase undotted-i is also wrong. It's correctly dotless but it is not a capital shape at all, that should be mapped to the same capital a small caps i would give in English.

**Comment by psb1558** [comment]
> Come to think of it, the lowercase undotted-i is also wrong. It's correctly dotless but it is not a capital shape at all, that should be mapped to the same capital a small caps i would give in English.

You mean when smcp is active? If so, I see I got the second case wrong not only here, but also in Junicode.

**Comment by alerque** [comment]
Yes, lowercase dotless-i (ı) when rendered with `+smcp` is not returning a small caps glyph, it is just the same one as it was. Note the 3rd letter in `Ilık`. Elstob is showing the regular lower case shape. The shape that should be shown is the same glyph as used in English for a capital I without a dot, but small cap sized of course.

**Comment by psb1558** [comment]
Thanks. That's what I thought, but I wanted to make sure it was straight in my head. Fixing now.

**Comment by psb1558** [comment]
I've uploaded just the variable fonts. Before I make a release, would you mind grabbing them and making sure I've got them right?

**Comment by psb1558** [comment]
I've uploaded new copies of the variable fonts after checking out the capitals with c2sc and fixing (or trying to fix) the fi and ffi ligatures. If you could check those things out, I'd be very grateful. In particular, a dotted i following f keeps its dot, and the narrow form of f is substituted. I'm pretty sure this is right:
<img width="137" alt="image" src="https://github.com/psb1558/Elstob-font/assets/16215767/3bde7a65-8f82-41c5-9f45-066bd2aad7a7">
But what should happen with dotless i? Should it be like this
<img width="51" alt="image" src="https://github.com/psb1558/Elstob-font/assets/16215767/2c77df06-6d2b-45aa-80be-96405e7af7dc">
or more like this (looking less like a ligature)?
<img width="53" alt="image" src="https://github.com/psb1558/Elstob-font/assets/16215767/b37dc1ee-278f-425f-aef0-d60b6c35c859">

**Comment by alerque** [comment]
Using the variable font build from cea8879cd05e2ded99a1fe9d4e404694467c50ff the results look correct. Updated test case using a local font file directly:

```sile
\begin[papersize=a7]{document}
\nofolios
\font[filename=./Elstob.ttf,language=en,features=+smcp]{First Input: }
\font[filename=./Elstob.ttf,language=tr,features=+smcp]{Bir İlk Ilık}

\font[family=Brill,language=en,features=+smcp]{First Input: }
\font[family=Brill,language=tr,features=+smcp]{Bir İlk Ilık}
\end{document}
```

![20231106_13h03m48s_grim](https://github.com/psb1558/Elstob-font/assets/173595/edca5abb-b311-4179-815f-4c721c27d7b5)

**Comment by psb1558** [comment]
Okay, thanks for testing. I'll update the repository and prepare a release sometime today.
